flink-connectors/flink-connector-files/src/main/java/org/apache/flink/connector/file/sink/committer/FileCommitter.java

```diff
@@ -48,9 +47,10 @@ public FileCommitter(BucketWriter<?, ?> bucketWriter) {
     }
 
     @Override
-    public List<FileSinkCommittable> commit(List<FileSinkCommittable> committables)
-            throws IOException {
-        for (FileSinkCommittable committable : committables) {
+    public void commit(Collection<CommitRequest<FileSinkCommittable>> requests)
+            throws IOException, InterruptedException {
+        for (CommitRequest<FileSinkCommittable> request : requests) {
```

Comment:
Perhaps we could handle the exceptions with `request.signalFailedWithKnownReason(e)` and other methods if it throws exception, or we create a follow-up issue to optimize the behavior of committing.

Reply:
This is inherited from the interface. Maybe we can discuss whether to allow throwing exceptions  here or to make a agreement that all exceptions while committing should be reported via the CommitRequest (unless the exception is not bounded to a specific request)?

Reply:
I think the key is that we might need to know the result of a single request. We might first not change the behavior here and open a follow-up issue for this part if there are still controversy~